staging/src/k8s.io/kubectl/pkg/cmd/auth/whoami.go

```diff
@@ -0,0 +1,174 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	authenticationv1alpha1 "k8s.io/api/authentication/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/printers"
+	authenticationv1alpha1client "k8s.io/client-go/kubernetes/typed/authentication/v1alpha1"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/scheme"
+	"k8s.io/kubectl/pkg/util/templates"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/yaml"
+)
+
+// WhoAmIOptions is the start of the data required to perform the operation. As new fields are added, add them here instead of
+// referencing the cmd.Flags()
+type WhoAmIOptions struct {
+	AuthClient authenticationv1alpha1client.AuthenticationV1alpha1Interface
+
+	TablePrinter printers.ResourcePrinter
+	OutputFormat *string
+
+	genericclioptions.IOStreams
+	PrintFlags *genericclioptions.PrintFlags
+}
+
+var (
+	whoAmILong = templates.LongDesc(`
+		Check who you are and your attributes (groups, extra).
+
+        This command is helpful to get yourself aware of the current user attributes, 
+        especially when dynamic authentication, e.g., token webhook, auth proxy, or OIDC provider, 
+        is enabled in the Kubernetes cluster.
+	`)
+
+	whoAmIExample = templates.Examples(`
+		# Get your subject attributes.
+		kubectl auth who-am-i
+		
+		# Get your subject attributes in JSON format.
+		kubectl auth who-am-i -o json
+	`)
+)
+
+// NewCmdWhoAmI returns an initialized Command for 'auth who-am-i' sub command
+func NewCmdWhoAmI(f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := &WhoAmIOptions{
+		OutputFormat: pointer.String(""),
+		IOStreams:    streams,
+		PrintFlags:   genericclioptions.NewPrintFlags("").WithTypeSetter(scheme.Scheme),
+	}
+
+	cmd := &cobra.Command{
+		Use:                   "who-am-i",
+		DisableFlagsInUseLine: true,
+		Short:                 "Check self subject attributes.",
+		Long:                  whoAmILong,
+		Example:               whoAmIExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(f, args))
+			cmdutil.CheckErr(o.RunAttributesGet())
+		},
+	}
+
+	allowedFormats := []string{"yaml", "json"}
+	cmd.Flags().StringVarP(o.OutputFormat, "output", "o", *o.OutputFormat, fmt.Sprintf(`Output format. One of: (%s).`, strings.Join(allowedFormats, ", ")))
+	return cmd
+}
+
+// Complete completes all the required options
+func (o *WhoAmIOptions) Complete(f cmdutil.Factory, args []string) error {
+	c, err := f.KubernetesClientSet()
```

Comment:
This is wrong, you should not be using `cmdutil.Factory` but rather `genericclioptions.RESTClientGetter` and afterwards `RESTClientGetter.ToRESTConfig` which will allow you to create the necessary config to access API. See https://github.com/kubernetes/kubernetes/blob/5a156349f6fdd31857249c30ebded7fcf19d30cf/staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go#L154-L162 as a perfect example how this should be done.

Reply:
Ok, I get it. It turned out that the code I used as an example (can-i command) was too outdated.
This one is fixed now.